### PR TITLE
Logging Plugin. (Fixes #104, Fixes #180)

### LINKED ISSUE #104: Logging Plugin
This is a plugin which automatically will record logging events.

See:
 * https://github.com/iopipe/iopipe-python/pull/211
 * https://github.com/iopipe/iopipe-python/issues/178
 * https://github.com/iopipe/iopipe-python/pull/185
 * https://github.com/iopipe/iopipe-go/issues/29

### LINKED ISSUE #180: Rename and deprecate old method specifying the profiler signer URL.
The URL used for the signer is _very_ specific to the profiler in that all the configuration stuff uses the word "Profiler" in it, this is not really the case as the signer can be used for anything. So to remove confusion this will be changed to just signer.

### PR DESCRIPTION
**Pull request by XerTheSquirrel**
This will implement the logging plugin.

### CODE REVIEW

**Comment by XerTheSquirrel** [comment]
Okay so requirements that I am putting down for the implementation of this in Java:

 * Internal logging API which gathers and generates reports to IOpipe.
   * Easily callable architecture for adding message with a logging level and message data.
 * Support the standard `java.util.logging` interfaces, can be built in since it always exists.
 * Support TinyLog, in a separate repository that way we can change the logging framework the agent uses itself and we will not be boxing ourselves in with this plugin and causing an issue later in time.
 * Support log4j2, in a separate repository so we do not force it to be brought in.

Questions to ask and go through:

 * Should logging be enabled by default?
 * Do we want to take over `System.out` and `System.err` and record all text which is printed to the console? This could potentially be intrusive, but CloudWatch will pick up all console output and generate messages accordingly. Should capturing these outputs be enabled by default, since there could be a gigantic ton of data printed to the console which might not even be useful, like fancy Spring Cloud Function logos. Note that standard output and standard error have no levels and some of their messages might be errors or not, do we parse strings or do we just assume a given stream is a given log level?
 * Store logging messages in memory (which can be a big one especially if memory counts are low and there are a few thousand messages and such) or do we want to use `/tmp`?
  * Store in a binary format that is simple to read?
  * Store a stream of data in how it is uploaded to IOpipe so no processing needs to be done?
 * Should `java.util.logging` support be enabled by default?

**Comment by XerTheSquirrel** [comment]
Will fix #180.

**Comment by XerTheSquirrel** [comment]
Ensure at the end that the split off signer code works in production, as it passes in tests.

**Comment by XerTheSquirrel** [comment]
Associated HUD issue for color keys: https://github.com/iopipe/hud/issues/1275

**Comment by XerTheSquirrel** [comment]
Consider SLF4J also.

**Comment by XerTheSquirrel** [comment]
Also consider Logback.

**Comment by XerTheSquirrel** [comment]
This is the base logging plugin and is ready for review, note that any integration with frameworks is separate! This is ready for review.

File: src/main/java/com/iopipe/plugin/logger/LoggerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/09/24
+	 */
+	@Override
+	public final String version()
+	{
+		return "1.9.0";
```
Comment (pselle) [inline src/main/java/com/iopipe/plugin/logger/LoggerPlugin.java]: Should this be 1.0?

File: src/main/java/com/iopipe/plugin/logger/LoggerExecution.java [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+					StandardOpenOption.CREATE,
+					StandardOpenOption.TRUNCATE_EXISTING,
+					StandardOpenOption.READ,
+					StandardOpenOption.WRITE);
+				
+				writer = new OutputStreamWriter(
+					Channels.newOutputStream(channel), "utf-8");
+			}
+			catch (IOException e)
+			{
```
Comment (pselle) [inline src/main/java/com/iopipe/plugin/logger/LoggerExecution.java]: Consider doing some kind of logging here, how are you going to catch issues if they arise?

File: src/main/java/com/iopipe/plugin/logger/LoggerPlugin.java (hunk 1 above)
Comment (XerTheSquirrel) [reply src/main/java/com/iopipe/plugin/logger/LoggerPlugin.java]: I will make this 1.0.

File: src/main/java/com/iopipe/plugin/logger/LoggerExecution.java (hunk 2 above)
Comment (pselle) [reply src/main/java/com/iopipe/plugin/logger/LoggerExecution.java]: After discussing, we'll hold on adding logging here out of recursive concerns -- if/when needs arise, we'll revisit where we need to add observability in here

**Comment by XerTheSquirrel** [comment]
Updated to 1.0.0.

**Pull request merged**